res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2415 @@
+// Sets the maximum adjustment of BPM allowed for beats to sync if AutoDJ is enabled [BPM].
+PioneerDDJSX.autoDJMaxBpmAdjustment = 10; //default: 10
+// If true, AutoDJ queue is being shuffled after skipping a track.
+// When using a fixed set of tracks without manual intervention, some tracks may be unreachable, 
+// due to having an unfortunate place in the queue ordering. This solves the issue.
+PioneerDDJSX.autoDJShuffleAfterSkip = false; //default: false
+
+// If true, by releasing rotary selector, 
+// track in preview player jumps forward to "jumpPreviewPosition". 
+PioneerDDJSX.jumpPreviewEnabled = true; //default: true
+PioneerDDJSX.jumpPreviewPosition = 0.3; //default: 0.3
+
+// If true, pad press in SAMPLER-PAD-MODE repeatedly causes sampler to play 
+// loaded track from cue-point, else it causes to play loaded track from the beginning.
+PioneerDDJSX.samplerCueGotoAndPlay = false; //default: false
+
+// If true, PFL / Cue (headphone) is being activated by loading a track into certain deck.
+PioneerDDJSX.autoPFL = true; //default: true
+
+// If true, new in Mixxx 2.1 introduced library controls will be used,
+// else old playlist controls will be used.
+PioneerDDJSX.useNewLibraryControls = false; //default: false
+
+// If true, SHIFT has to be pressed to activate needle search control
+PioneerDDJSX.needleSearchShiftEnable = false; //default: false
+
+
+///////////////////////////////////////////////////////////////
+//               INIT, SHUTDOWN & GLOBAL HELPER              //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSX.shiftPressed = false;
+PioneerDDJSX.rotarySelectorChanged = false;
+PioneerDDJSX.panels = [false, false]; // view state of effect and sampler panel
+
+PioneerDDJSX.syncRate = [0, 0, 0, 0];
+PioneerDDJSX.gridAdjustSelected = [false, false, false, false];
+PioneerDDJSX.gridSlideSelected = [false, false, false, false];
+PioneerDDJSX.needleSearchTouched = [false, false, false, false];
+PioneerDDJSX.chFaderStart = [null, null, null, null];
+PioneerDDJSX.toggledBrake = [false, false, false, false];
+PioneerDDJSX.scratchMode = [true, true, true, true];
+PioneerDDJSX.wheelLedsBlinkStatus = [0, 0, 0, 0];
+PioneerDDJSX.padMode = [0, 0, 0, 0]; //0 = HOTCUE, 1 = ROLL, 2 = SLICER, 3 = SAMPLER
+PioneerDDJSX.samplerVelocityMode = [false, false, false, false];
+
+// FX storage:
+PioneerDDJSX.fxKnobMSBValue = [0, 0];
+PioneerDDJSX.shiftFxKnobMSBValue = [0, 0];
+
+// used for advanced auto dj features:
+PioneerDDJSX.blinkAutodjState = false;
+PioneerDDJSX.autoDJTickTimer = 0;
+PioneerDDJSX.autoDJSyncBPM = false;
+PioneerDDJSX.autoDJSyncKey = false;
+
+// init helpers:
+PioneerDDJSX.initHandshakeTime = 0;
+PioneerDDJSX.initSendMsg = [
+    0xF0, 0x00, 0x20, 0x7F, 0x01, 0x02, 0x01, 0x01, 0x12,
+    0x06, 0x0D, 0x03, 0x09,
+    0x0A, 0x08, 0x03, 0x00,
+    0x05, 0x09, 0x0F, 0x0F, 0x01, 0x08, 0x00, 0x09, 0xF7
+];
+PioneerDDJSX.initRecvMsg = [
+    0xF0, 0x00, 0x20, 0x7F, 0x01, 0x02, 0x01, 0x02, 0x12,
```

Comment:
It's independent of the serial number, because I tested two different controllers of the same type. Firmware version....I'm not really sure, I just saw that every hex value after the first 7 changes every time serato starts, so I think it has something to do with timestamp and checksum.